src/DonationForms/Endpoints/Endpoint.php

```diff
@@ -28,11 +28,23 @@ public function validateInt($value)
 
     /**
      * Check user permissions
+     *
+     * @unreleased Check for different capabilities for different HTTP methods
+     * @since      2.19.0
+     *
      * @return bool|WP_Error
      */
     public function permissionsCheck()
     {
-        if ( ! current_user_can('edit_posts')) {
+        if (in_array($_SERVER['REQUEST_METHOD'], ['POST', 'UPDATE', 'DELETE'], true)) {
```

Comment:
What is the reason that we are checking for specific request methods before checking the nonce? My concern is that we are now using this method to handle multiple different cases, which may have unintended consequences when called.

Reply:
@kjohnson 

Once the nonce issue has been resolved (WP checks it before triggering the `permissionsCheck` method) maybe that resolves the question.

About the method itself, I'm trying to raise the bar. Currently, a contributor is able to delete entries through the API endpoint.

Your suggestions are more than welcome.

Reply:
@pauloiankoski it sounds to me like we need to be implementing the `permissionsCheck()` method on each of the endpoints, instead of trying to generalize in the abstract Endpoint class.